core/offchain.go

```diff
@@ -228,31 +228,26 @@ func (bc *BlockChain) CommitOffChainData(
 			); err != nil {
 				return NonStatTy, err
 			}
-			for _, paid := range [...][]reward.Payout{
-				roundResult.BeaconchainAward, roundResult.ShardChainAward,
-			} {
-				for i := range paid {
-					stats, ok := tempValidatorStats[paid[i].Addr]
-					if !ok {
-						stats, err = bc.ReadValidatorStats(paid[i].Addr)
-						if err != nil {
-							utils.Logger().Info().Err(err).
-								Str("addr", paid[i].Addr.Hex()).
-								Str("bls-earning-key", paid[i].EarningKey.Hex()).
-								Msg("could not read validator stats to update for earning per key")
-							continue
-						}
-						tempValidatorStats[paid[i].Addr] = stats
+			for _, paid := range roundResult.Payouts {
+				stats, ok := tempValidatorStats[paid.Addr]
```

Comment:
Note: no logic change within the loop. Only the loop is changed to a merged list.